### PR TITLE
fix: make WatchResourceTree use namespaced cache key

### PR DESCRIPTION
**Pull request by torfjor**
`WatchResourceTree` is using the non-namespaced cache key when watching namespaced applications. That ends up with us looking in the wrong place for the resource tree.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -1399,9 +1399,14 @@ func (s *Server) WatchResourceTree(q *application.ResourcesQuery, ws application
 		return err
 	}
 
-	return s.cache.OnAppResourcesTreeChanged(ws.Context(), q.GetApplicationName(), func() error {
+	cacheKey := q.GetApplicationName()
+	if ns := q.GetAppNamespace(); ns != "" && ns != s.ns {
+		cacheKey = fmt.Sprintf("%s_%s", ns, cacheKey)
+	}
```
Comment (jannfis) [inline server/application/application.go:1405]: Please use `AppInstanceName` from `util/argo` to get the cache key instead of rolling your own.

**Review by jannfis — CHANGES_REQUESTED** [review]
Great catch, thanks! I have an request for change though, PTAL

File: server/application/application.go (hunk 1 above)
Comment (torfjor) [reply server/application/application.go]: Didn't notice we had this. Thanks!

**Comment by jannfis** [comment]
Thank you. This PR is not associated with a bug issue. Is there a bug that this fixes? What is the impact of this change?

**Review by jannfis — APPROVED** [review]
LGTM! Thanks!

**Comment by torfjor** [comment]
@jannfis any idea when this will land in a release?

**Comment by jannfis** [comment]
@torfjor I don't see a bug or similar describing the impact of this fix. So for now, this will land in Argo CD 2.9.

**Comment by jannfis** [comment]
Ah! Sorry. Missed your comment. D'oh.

**Comment by jannfis** [comment]
So: This fix will end up in 2.8.5, 2.7.15 and 2.6.16.

**Pull request merged**